### PR TITLE
[MRG] Beautifying plots: dim-labels, log-scale.

### PR DESCRIPTION
**Pull request by jusjusjus**
See discussion history in #518 and #517.

I rebased to upstream/master and updated the code according to your suggestions.  (sorry for starting yet another pull request.  I always get entangled in git issues when trying to rebase.)

@glouppe :  Of what would you like me to generate screenshots?

Regarding `space.Categorical`, We could construct plottable dimensions along these lines:
`dims_to_plot = [not isinstance(dim, Categorical) for dim in space.dimensions]`, and go from there.  I think the property that we're missing is something like a `Dimensions.has_ordering`.  What do you think?

Updated use example:
```
import matplotlib.pyplot as plt
from skopt import forest_minimize
from skopt.plots import plot_evaluations, plot_objective
from skopt.benchmarks import branin as branin
bounds = [
    (-5.0, 10.0), 
    (0.01, 15.0, 'log-uniform')
]
res = forest_minimize(branin, bounds, n_calls=25, base_estimator="ET",
                             random_state=42)

res.space.dimensions[0].name = 'Dim 0'
plot_evaluations(res, bins=10)
plt.show()
```

### CODE REVIEW

File: skopt/plots.py [hunk 1]
```diff
@@ -118,15 +122,27 @@ def _format_scatter_plot_axes(ax, space, ylabel):
...
                 else:
                     [l.set_rotation(45) for l in ax_.get_xticklabels()]
-                    ax_.set_xlabel("$X_{%i}$" % j)
+                    ax_.set_xlabel(dim_labels[j])
+                
+                # configure plot for linear vs log-scale
+                priors = (space.dimensions[j].prior, space.dimensions[i].prior)
+                scale_setters = (ax_.set_xscale, ax_.set_yscale)
+                loc_setters = (ax_.xaxis.set_major_locator,
+                        ax_.yaxis.set_major_locator)
```
Comment (betatim) [inline skopt/plots.py]: nitpick: can you indent this so the two `ax_`s are at the same level please

File: skopt/plots.py [hunk 2]
```diff
@@ -118,15 +122,27 @@ def _format_scatter_plot_axes(ax, space, ylabel):
...
+                for set_major_locator, set_scale, prior in zip(loc_setters,
+                        scale_setters, priors):
```
Comment (betatim) [inline skopt/plots.py]: indent this so it lines up with `loc_setters` or make all three argument be on a new line?

File: skopt/plots.py [hunk 3]
```diff
@@ -378,4 +410,5 @@ def plot_evaluations(result, bins=20):
                 ax[i, j].scatter(result.x[j], result.x[i],
                                  c=['r'], s=20, lw=0.)
 
-    return _format_scatter_plot_axes(ax, space, "Number of samples")
+    return _format_scatter_plot_axes(ax, space, ylabel="Number of samples",
+            dim_labels=dimensions)
```
Comment (betatim) [inline skopt/plots.py]: indent me to line up with `ax`

**Comment by betatim** [comment]
Let's merge this after the nitpicks on style.

**Comment by jusjusjus** [comment]
I fixed all the nitpicks (I hope) and rebased to the current master.

**Comment by betatim** [comment]
Thank you :)

**Comment by glouppe** [comment]
Thanks a lot for your contribution @jusjusjus !

**Pull request merged**